pkgs/os-specific/darwin/installApplication/default.nix

```diff
@@ -0,0 +1,43 @@
+{ cpio, fetchurl, fixDarwinDylibNames, lib, stdenvNoCC, undmg, unpkg, unzip }:
+{ description, homepage, license, maintainers, pname, sha256, url, version }:
+
+stdenvNoCC.mkDerivation rec {
+  inherit pname version;
+
+  nativeBuildInputs = [ cpio fixDarwinDylibNames undmg unpkg unzip ];
+
+  sourceRoot = ".";
+  src = fetchurl {
+    name = builtins.replaceStrings [ "%20" ] [ "-" ] (builtins.head (builtins.match ".*/([^/]+)" url));
+    inherit url sha256;
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+
+  installPhase = ''
+      # .dmg files or compressed Applications
+      app=( ./*.app )
+      if [ ! -z "$app" ]; then
+        mkdir -p $out/Applications
+        mv -n "$app" $out/Applications/
```

Comment:
If I read this bash correctly, then `app=( ./*.app )` creates an array of all application bundles, while `mv -n "$app" $out/Applications/` will move just the first one over to `$out/Applications`